game.cpp

```diff
@@ -717,6 +750,39 @@ void test_game() //!OCLINT tests may be many
       }
   }
 
+  // Player enters cool down status after one shooting action
+  {
+    game g;
+    g.do_action(0, action_type::shoot);
+    g.tick();
+    assert(g.get_player(0).is_shoot_cool_down());
+    g.do_action(0, action_type::shoot);
+    assert(!g.get_player(0).is_shooting());
+  }
+
+  // Player can shoot again after the interval
+  {
+    game g;
+    g.do_action(0, action_type::shoot);
+    for (auto i = 0; i <= projectile::m_fire_rate / g.get_player(0).get_shoot_fire_rate_multiplier(); ++i)
+      {
+        g.tick();
```

Comment:
I feel a welcome (but optional) bonus would be to check that the player still cannot shoot during any of these ticks (except the last one I think?)

Reply:
I was thinking about it, I did not implement this because I believe the case was already covered in the first test. But of course, adding this will further strengthen the safe net, I will add it to the test

Reply:
@EvoLandEco the rule is that a test should break the code. If it did not, the test can be deleted. And yes, there will be tests that overlap in testing the same thing. You may delete some obsolete tests, just make sure it is reviewed :-)